pinot-segment-local/src/main/java/org/apache/pinot/segment/local/recordtransformer/CompositeTransformer.java

```diff
@@ -120,6 +120,9 @@ record = transformer.transform(record);
       if (record == null) {
         return null;
       }
+      if (Boolean.TRUE.equals(record.getValue(GenericRow.SKIP_RECORD_KEY))) {
```

Comment:
We should move this to line 118 (before the for loop). We may use `IngestionUtils.shouldIngestRow(record)`

Reply:
Hi @Jackie-Jiang . I don't find any other operations which could mark record as SKIP_RECORD instead of `FilterTransformer`. And `FilterTransformer` is the second transformer of `CompositeTransformer`. I think we should put this predicate in the for loop. Could you help review again?

Reply:
I see. In that case I'd suggest moving the check to the first line in the for loop (before calling `transformer.transform()`) and only check `if (!IngestionUtils.shouldIngestRow(record))` to make the handling more general. This special key might be added from the upstream in certain scenario (code might not be managed in Pinot).